backend/src/Designer/Controllers/DatamodelsController.cs

```diff
@@ -248,6 +249,27 @@ public async Task<IActionResult> UseXsdFromRepo(string org, string repository, s
             }
         }
 
+        /// <summary>
+        /// Gets the dataType for a given datamodel.
+        /// </summary>
+        [HttpGet("datamodel/{modelName}/dataType")]
+        [UseSystemTextJson]
+        public async Task<IActionResult> GetModelDataType(string org, string repository, string modelName)
+        {
+            DataType dataTypeConfiguration = await _schemaModelService.GetModelDataType(org, repository, modelName);
+            return Ok(dataTypeConfiguration);
+        }
+
+        /// <summary>
+        /// Updates the dataType for a given model.
+        /// </summary>
+        [HttpPut("datamodel/{modelName}/dataType")]
```

Comment:
Can we remove the modelName param from the route?

Reply:
After thinking it over, I updated the method to check that the ID in the URI matches the one in the request body. If they don’t match, it now returns a Bad Request response.
This aligns with the HTTP standard and RESTful design principles, where the URI is expected to uniquely identify the resource for a PUT request.
I’ve also added a guard for this validation and written a test specifically for this case.